azurerm/resource_arm_sql_firewall_rule.go

```diff
@@ -51,96 +50,86 @@ func resourceArmSqlFirewallRule() *schema.Resource {
+
+	if result.ID == nil {
+		return fmt.Errorf("Cannot create sql firewall rule %s (resource group %s) ID", name, resGroup)
 	}
 
-	resp := readResponse.Parsed.(*sql.GetFirewallRuleResponse)
-	d.SetId(*resp.ID)
+	d.SetId(*result.ID)
 
 	return resourceArmSqlFirewallRuleRead(d, meta)
 }
 
 func resourceArmSqlFirewallRuleRead(d *schema.ResourceData, meta interface{}) error {
+	sqlFirewallRulesClient := meta.(*ArmClient).sqlFirewallRulesClient
+
 	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
 		return err
 	}
-	resGroup := id.ResourceGroup
 
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
-
-	readRequest := rivieraClient.NewRequestForURI(d.Id())
-	readRequest.Command = &sql.GetFirewallRule{}
+	resGroup := id.ResourceGroup
+	name := id.Path["firewallRules"]
+	serverName := id.Path["servers"]
 
-	readResponse, err := readRequest.Execute()
+	result, err := sqlFirewallRulesClient.Get(resGroup, serverName, name)
 	if err != nil {
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", err)
+		return fmt.Errorf("Error reading SQL firewall rule %s: %v", name, err)
 	}
-	if !readResponse.IsSuccessful() {
-		log.Printf("[INFO] Error reading SQL Server Firewall Rule %q - removing from state", d.Id())
+	if result.Response.StatusCode == http.StatusNotFound {
 		d.SetId("")
-		return fmt.Errorf("Error reading SQL Server Firewall Rule: %s", readResponse.Error)
+		return nil
 	}
 
-	resp := readResponse.Parsed.(*sql.GetFirewallRuleResponse)
+	if props := result.FirewallRuleProperties; props != nil {
+		d.Set("start_ip_address", props.StartIPAddress)
+		d.Set("end_ip_address", props.EndIPAddress)
+	}
 
+	d.Set("name", name)
 	d.Set("resource_group_name", resGroup)
-	d.Set("location", azureRMNormalizeLocation(*resp.Location))
-	d.Set("name", resp.Name)
-	d.Set("server_name", id.Path["servers"])
-	d.Set("start_ip_address", resp.StartIPAddress)
-	d.Set("end_ip_address", resp.EndIPAddress)
+	d.Set("server_name", serverName)
 
 	return nil
 }
 
 func resourceArmSqlFirewallRuleDelete(d *schema.ResourceData, meta interface{}) error {
-	client := meta.(*ArmClient)
-	rivieraClient := client.rivieraClient
+	sqlFirewallRulesClient := meta.(*ArmClient).sqlFirewallRulesClient
 
-	deleteRequest := rivieraClient.NewRequestForURI(d.Id())
-	deleteRequest.Command = &sql.DeleteFirewallRule{}
-
-	deleteResponse, err := deleteRequest.Execute()
+	id, err := parseAzureResourceID(d.Id())
 	if err != nil {
-		return fmt.Errorf("Error deleting SQL Server Firewall Rule: %s", err)
+		return err
 	}
-	if !deleteResponse.IsSuccessful() {
-		return fmt.Errorf("Error deleting SQL Server Firewall Rule: %s", deleteResponse.Error)
+
+	resGroup := id.ResourceGroup
+	name := id.Path["firewallRules"]
+	serverName := id.Path["servers"]
+
+	result, error := sqlFirewallRulesClient.Delete(resGroup, serverName, name)
+	if result.Response.StatusCode != http.StatusOK {
+		return fmt.Errorf("Error deleting sql firewall rule %s: %+v", name, error)
```

Comment:
given the swagger status codes can change over time (200 -> 201 etc) - can we change this to check for an error and within that handle the not found case using the wrapper method? [Here's an example from the SQL Server resource](https://github.com/terraform-providers/terraform-provider-azurerm/blob/2c73509a1b276a4439c9da5d722ffd59bb011a9e/azurerm/resource_arm_sql_server.go#L157)